README.md

```diff
@@ -2,7 +2,51 @@
 
 **This is not an officially supported Google product.**
 
-Protein database search tools such as BLAST are instrumental for research in the life sciences. However, they are slow and are based on surface-level sequence similarity. We are exploring using neural networks to improve the speed and accuracy of finding relevant sequences from these databases.
+Protein database search tools such as BLAST are instrumental for research in the life sciences. However, they are slow and are based on surface-level sequence similarity. We are exploring using neural networks to improve the speed and accuracy of finding relevant sequences from these databases. More specifically, we are aiming to learn fixed-length protein embeddings using contextual lenses (https://arxiv.org/pdf/2002.08866.pdf). Generally speaking, a sequence level protein representation, such as a one-hot encoding, is an array of the the form (sequence_length, n) where n is the amino acid embedding dimension. A contextual lens is a (learnable) map from the (sequence_length, n)-array to an (m,)-vector where m is independent of sequence_length. Embeddings are constructed using an encoder function followed by a contextual lens. To learn these embeddings a downstream prediction task is performed using a single feedforward layer. Gradients are backpropagated through all 3 components of the architecture (encoder, lens, predictor) using the Adam optimizer with variable (potentially zero) learning rates per component.
+
+### Encoders
+- One-hot: non-learnable
+- CNN: learnable
+- Transformer: learnable and pretrainable
+
+### Lenses
+- Mean/Max-Pool: non-learnable
+- Linear-Mean/Max-Pool: learnable
+- GatedConvolution: learnable and self-attentive
+
+## Downstream Task
```

Comment:
what about the TAPE stuff? Should you mention that?

Reply:
good question, the TAPE stuff is done in an entirely separate colab notebook, I can add that here and refer to it?